crates/static-analysis-kernel/src/analysis/analyze.rs

```diff
@@ -29,6 +29,51 @@ fn get_lines_to_ignore(code: &str, language: &Language) -> LinesToIgnore {
     let mut lines_to_ignore_per_rules: HashMap<u32, Vec<String>> = HashMap::new();
 
     let mut line_number = 1u32;
+    let comment_starters = match language {
+        Language::Python
+        | Language::Starlark
+        | Language::Dockerfile
+        | Language::Elixir
+        | Language::Ruby
+        | Language::Terraform
+        | Language::Yaml
+        | Language::Bash
+        | Language::R => {
+            vec!["#"]
+        }
+        Language::JavaScript | Language::TypeScript | Language::Kotlin | Language::Apex => {
+            vec!["//", "/*"]
+        }
+        Language::Go | Language::Rust | Language::Csharp | Language::Java | Language::Swift => {
+            vec!["//"]
+        }
+        Language::Json => {
+            vec!["impossiblestringtoreach"]
+        }
+        Language::PHP => {
+            vec!["//", "/*", "#"]
+        }
+        Language::Markdown => {
+            vec!["<!-"]
+        }
+        Language::SQL => {
+            vec!["--", "/*"]
+        }
+    };
```

Comment:
We're sorta duplicating information in two match blocks, one here and one later on when we compute "disabling_patterns" - why don't we have a look-up map for language -> comment prefix, and then have `disabling_patterns` be computed independent of the language (removing the "duplicate" match), e.g. something like

```rs
["no-dd-sa", "datadog-disable"]
    .iter()
    .map(|pattern| 
        comment_prefixes // e.g. ["//", "/*"]
            .iter()
            .map(|prefix| prefix + pattern) // gives us a 2x2 matrix of all possible patterns, so we'll flatten it
    ).flatten()
```